src/main/java/com/cedarsoftware/util/io/ObjectResolver.java

```diff
@@ -678,6 +697,14 @@ else if (element instanceof JsonObject)
         jsonObj.clearArray();
     }
 
+    protected static boolean isNonEmptyString(Object element) {
+        return element instanceof String && "".equals(((String) element).trim());
+    }
+
+    protected static boolean isCompTypeStringOrObject(final Class compType) {
```

Comment:
Maybe this should be `isComponentTypeOfClass(Class...classes)` and possibly be in MetaUtils since it is a generic method that is not spcecific to ObjectResolver.